### PR TITLE
COM-12460: add ISO_IEC_13818-4 MPEG-2 AAC compressedAdif test suite

### PR DESCRIPTION
**Pull request by rsanchez87**

### CODE REVIEW

**Review by mdimopoulos — COMMENTED** [review]
A couple of points to clarify.

File: scripts/gen_aac.py [hunk 1]
```diff
@@ -227,7 +244,12 @@ def generate(self, download, jobs):
         self._download_raw_output_references_and_checksums(jobs, test_suite, raw_bitstream_links,
                                                            raw_bitstream_md5_links)
 
-        for test_vector in test_suite.test_vectors.values():
+        valid_test_vectors = [
+            test_vector for test_vector in test_suite.test_vectors.values()
+            if test_vector.name not in missing_checksum_files
+        ]
+
+        for test_vector in valid_test_vectors:
```
[the remark this answers is not in the document]
Comment (rsanchez87) [reply scripts/gen_aac.py:252]: It should be the same, but I chose **.md5** thinking in **_fill_checksum** function, but yes, we can talk about this, thanks

File: fluster/utils.py [hunk 2]
```diff
@@ -18,24 +18,41 @@
...
+def download(url: str, dest_dir: str, max_retries: int = 5) -> None:
+    """Downloads a file to a directory with a mutex lock to avoid conflicts and retries with exponential backoff."""
+    for attempt in range(max_retries):
+        try:
+            with download_lock:
+                with urllib.request.urlopen(url) as response:
+                    dest_path = os.path.join(dest_dir, url.split("/")[-1])
+                    with open(dest_path, "wb") as dest:
+                        shutil.copyfileobj(response, dest)
+            print(f"Download completed for {url}")
```
[the remark this answers is not in the document]
Comment (rsanchez87) [reply fluster/utils.py:47]: Yes, thanks, it's not necessary, it was to debug...

[the remark this answers is not in the document]
Comment (rsanchez87) [reply fluster/utils.py]: Done

File: test_suites/aac/MPEG2_AAC-ADIF.json
[the remark this answers is not in the document]
Comment (rsanchez87) [reply test_suites/aac/MPEG2_AAC-ADIF.json:1]: Done

**Review by mdimopoulos — APPROVED** [review]

**Pull request merged**